src/main/java/io/confluent/kafkarest/resources/TopicsResource.java

```diff
@@ -74,6 +58,37 @@ public Topic getTopic(@PathParam("topic") String topicName) {
     return topic;
   }
 
+  @GET
+  @Path("/{topic}/partition/{partition}/messages")
+  @PerformanceMetric("topic.consume-binary")
+  @Produces({Versions.KAFKA_V1_JSON_BINARY_WEIGHTED,
+             Versions.KAFKA_V1_JSON_WEIGHTED,
+             Versions.KAFKA_DEFAULT_JSON_WEIGHTED,
+             Versions.JSON_WEIGHTED,
+             Versions.KAFKA_V1_JSON_BINARY_WEIGHTED_LOW,
+             Versions.ANYTHING})
+  public void consumeBinary(final @Suspended AsyncResponse asyncResponse,
+                            final @PathParam("topic") String topicName,
+                            final @PathParam("partition") int partitionId,
+                            final @QueryParam("offset") long offset,
+                            final @QueryParam("count") @DefaultValue("1") long count) {
```

Comment:
A default value of 1 here (and on the avro version) seems like a bad idea since it seems unlikely to almost ever be the right choice. The only scenario I can come up with where that's a useful value is for debugging, i.e. you know the offset of a particular message that you want to see the contents of.

I had a similar problem when trying to decide how many records to return in a single response for the high-level consumer interface. Messages turned out to just be a bad unit to use because messages can vary pretty widely in size depending on use case. Setting any default ended up either causing really low throughput (when messages are tiny) or really huge responses (when messages were too large).

I can think of a couple of options. The first is to not even set a default. Require them to explicitly decide how much they want. The second option is to use # of bytes instead. That works if you're actually trying to consume from the stream normally and want reasonably sized chunks. A third option is the combination -- make them individually optional, but at least one must be specified and if both are, they are both enforced.